### PR TITLE
BUG: fix df concat containing mix of localized and non-localized Timestamps

### PR DESCRIPTION
**Pull request by tsdlovell**
BUG: fix issue with concat of localized timestamps
TST: test concat of dataframes with non-None timezone columns

### CODE REVIEW

File: pandas/tests/frame/test_combine_concat.py [hunk 1]
```diff
@@ -46,6 +46,27 @@ def test_combine_multiple_frames_dtypes(self):
         expected = Series(dict(float64=2, float32=2))
         assert_series_equal(results, expected)
 
+    def test_combine_multiple_tzs(self):
+        ts1 = Timestamp('2015-01-01', tz=None)
```
Comment (jreback) [inline pandas/tests/frame/test_combine_concat.py]: add the issue number here as well

**Comment by tsdlovell** [comment]
Created #12467 and added it in test comment

**Comment by jreback** [comment]
does this close #12396 or #12403 ?

**Comment by jreback** [comment]
ping!

**Comment by tsdlovell** [comment]
Hey Jeff, I've looked into the other failures a little but don't fully understand when/why the expected dims change.  It looks like merge (the tests that fail) always wants a 1d array and is failing because we are using atleast_2d on one of our inputs to make our (2-d, 1-d) input combo compatible.  Perhaps we should instead be making the 2-d into a 1-d?  Is this just playing wack-a-mole?

**Comment by jreback** [comment]
a bit of a hack, but assure 2d-ness only when we are axis=1 concat.

```
diff --git a/pandas/tseries/common.py b/pandas/tseries/common.py
index f32e9b9..11a5fdc 100644
--- a/pandas/tseries/common.py
+++ b/pandas/tseries/common.py
@@ -260,8 +260,7 @@ def _concat_compat(to_concat, axis=0, typs=None):
         # if dtype is of datetimetz or timezone
         if x.dtype.kind == _NS_DTYPE.kind:
             if getattr(x, 'tz', None) is not None:
-                x = x.asobject
-                x = np.atleast_2d(x)
+                x = x.asobject.values
             else:
                 shape = x.shape
                 x = tslib.ints_to_pydatetime(x.view(np.int64).ravel())
@@ -272,6 +271,8 @@ def _concat_compat(to_concat, axis=0, typs=None):
             x = tslib.ints_to_pytimedelta(x.view(np.int64).ravel())
             x = x.reshape(shape)
 
+        if axis == 1:
+            x = np.atleast_2d(x)
         return x
 
     if typs is None:
```

**Comment by tsdlovell** [comment]
That passes the tests that were failing. Unfortunately, can't push from work.  Will make that change this evening.

**Comment by jreback** [comment]
@tsdlovell np.

**Comment by tsdlovell** [comment]
travis is failing on OSX and py3.5 at the same points

> ERROR: test_range_slice_day (pandas.tseries.tests.test_period.TestPeriodIndex)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/travis/build/pydata/pandas/pandas/tseries/tests/test_period.py", line 2619, in test_range_slice_day
    idx[v:]
  File "/Users/travis/build/pydata/pandas/pandas/tseries/base.py", line 176, in __getitem__
    result = getitem(key)
IndexError: failed to coerce slice entry of type str to integer
======================================================================
ERROR: test_range_slice_seconds (pandas.tseries.tests.test_period.TestPeriodIndex)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/travis/build/pydata/pandas/pandas/tseries/tests/test_period.py", line 2676, in test_range_slice_seconds
    idx[v:]
  File "/Users/travis/build/pydata/pandas/pandas/tseries/base.py", line 176, in __getitem__
    result = getitem(key)
IndexError: failed to coerce slice entry of type str to integer

Was working in the env I created a bit ago, but fails in an env I just created, so maybe something to do with change in dependencies.   Will look into it.

**Comment by jreback** [comment]
numpy 1.11 was just updated fully in conda
I am about to push the fixes for this

**Comment by jreback** [comment]
rebase and should be good to go

**Comment by jreback** [comment]
@tsdlovell thank you sir!

**Pull request closed without merging**